lib/simplecov.rb

```diff
@@ -79,9 +79,9 @@ def result
       # first, then merge the results and return those
       if use_merging
         SimpleCov::ResultMerger.store_result(@result) if @result
-        return SimpleCov::ResultMerger.merged_result
-      else
-        return @result if defined? @result
+        SimpleCov::ResultMerger.merged_result
+      elsif defined? @result
+        @result if defined? @result
```

Comment:
Reading this method makes me dizzy, maybe we get to untangle this a bit:

```ruby
def result
  # Ensure the variable is defined to avoid ruby warnings
  @result = nil unless defined?(@result)

  # Collect our coverage result
  if running and not result?
    @result = SimpleCov::Result.new add_not_loaded_files(Coverage.result)
  end

  # If we're using merging of results, store the current result
  # first, then merge the results and return those
  if use_merging
    if result?
      SimpleCov::ResultMerger.store_result @result
    end
    SimpleCov::ResultMerger.merged_result
  else
    @result
  end
ensure
  self.running = false
end
```